### PR TITLE
Fixes #23067 - expose option to use oVirt APIv4

### PR DESCRIPTION
**Pull request by iNecas**
Requires https://github.com/fog/fog-ovirt/pull/10

### CODE REVIEW

**Comment by orrabin** [comment]
I tested with fog/fog-ovirt#10 and it works well.

**Comment by iNecas** [comment]
The fog-ovirt part is merged, I've bumped the version dependency on fog-ovirt, this should be ready for the final review.

**Comment by iNecas** [comment]
And btw. I've tested the migration and works like a charm.

**Comment by orrabin** [comment]
Tested again, this time with image based provisioning - works well.
Tests need to run but :+1: from me to merge when tests are green.

**Comment by iNecas** [comment]
[test foreman]

**Comment by iNecas** [comment]
[test katello] [test upgrade]

**Comment by iNecas** [comment]
The tests are :green_heart: now

**Review by orrabin — DISMISSED** [review]

**Comment by mmoll** [comment]
@houndci-bot please have a look into our .rubocop*.yml, thanks

**Review by orrabin — APPROVED** [review]

File: app/models/compute_resources/foreman/model/ovirt.rb [hunk 1]
```diff
@@ -101,6 +101,21 @@ def max_memory
       16.gigabytes
     end
 
+    def use_v4=(value)
+      value = case value
```
Comment (ekohl) [inline app/models/compute_resources/foreman/model/ovirt.rb:105]: Isn't `value = [true, '1'].include?(value)` equivalent but shorter? Or is that less readable?

**Review by ekohl — APPROVED** [review]
Will need https://github.com/theforeman/foreman-packaging/pull/2351

File: app/models/compute_resources/foreman/model/ovirt.rb (hunk 1 above)
Comment (iNecas) [reply app/models/compute_resources/foreman/model/ovirt.rb:105]: Honestly, this option haven't occurred to me when writing this. The current variant seems a bit more explicit, but I would prefer this way mostly to not having to wait to tests to finish again, especially as today's the 1.18 branching day and we would like to get this in to the release.

**Comment by orrabin** [comment]
Merged! Thanks @iNecas!

**Comment by orrabin** [comment]
Thanks to @mmoll and @ekohl as well :)

**Comment by ares** [comment]
Thanks @orrabin for merging, please don't forget to set the release in redmine issue after merging.

**Pull request merged**